### PR TITLE
feat: Added JSON check.

### PR DESCRIPTION
**Pull request by sonnykt**
```yaml
json:
  - name: Validate composer.json
    file: composer.json
    ignore-missing: true
    key-values:
      - key: '$.repositories..no-api'
        optional: true
        is-list: true
        disallowed-values:
          - true
          - "true"
          - 1
          - "1"
        allowed-values:
          - false
          - "false"
          - 0
          - "0"
          -
```

### CODE REVIEW

**Comment by sonnykt** [comment]
@steveworley docker build is failing with error:
```
#9 0.582 go: go.mod file indicates go 1.21, but maximum version supported by tidy is 1.20
#9 ERROR: process "/bin/sh -c GOOS=${TARGETOS} GOARCH=${TARGETARCH} &&     go mod tidy &&     go generate ./... &&     go build -ldflags=\"-s -w -X main.version=${VERSION} -X main.commit=${COMMIT}\" -o build/shipshape" did not complete successfully: exit code: 1
```

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -22,7 +22,7 @@ jobs:
     - name: Set up Go
       uses: actions/setup-go@main
       with:
-        go-version: 1.17
+        go-version: 1.21
```
Comment (steveworley) [inline .github/workflows/go.yml:25]:
```suggestion
        go-version: 1.20
```
```
go: go.mod file indicates go 1.21, but maximum version supported by tidy is 1.20
```

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -62,7 +62,7 @@ jobs:
...
```
Comment (steveworley) [inline .github/workflows/go.yml:65]:
```suggestion
        go-version: 1.20
```

**Review by steveworley — CHANGES_REQUESTED** [review]

**Comment by steveworley** [comment]
Hey @sonnykt yeah `go mod tidy` doesn't look to support 1.21 just yet — so we'll have to pin to 1.20.

File: go.mod [hunk 3]
```diff
@@ -1,10 +1,12 @@
 module github.com/salsadigitalauorg/shipshape
 
-go 1.17
+go 1.21
```
Comment (yusufhm) [inline go.mod:3]: Another one here for pinning the go version:
```suggestion
go 1.20
```

**Comment by sonnykt** [comment]
@steveworley turned out that we were using Docker image 1.20. I've increased the version to 1.21 and all good now.

**Review by steveworley — APPROVED** [review]

**Pull request merged**